lib/swaps/Swaps.ts

```diff
@@ -1147,11 +1139,19 @@ class Swaps extends EventEmitter {
 
     clearTimeout(this.timeouts.get(deal.rHash));
     this.timeouts.delete(deal.rHash);
-    if (deal.phase === SwapPhase.SwapAccepted && deal.role === SwapRole.Maker) {
-      // if we are the maker and we have accepted a swap deal but we haven't yet started paying the taker
-      // then we should cancel
-      const swapClient = this.swapClientManager.get(deal.role === SwapRole.Maker ? deal.makerCurrency : deal.takerCurrency)!;
-      swapClient.removeInvoice(deal.rHash).catch(this.logger.error); // we don't need to await the remove invoice call
+    if (deal.role === SwapRole.Maker) {
+      if (deal.phase === SwapPhase.SwapAccepted) {
+        // if we are the maker and we have accepted a swap deal but we haven't yet started paying the taker
+        // then we should cancel the invoice for our incoming payment if one exists
+        const swapClient = this.swapClientManager.get(deal.makerCurrency)!;
+        swapClient.removeInvoice(deal.rHash).catch(this.logger.error); // we don't need to await the remove invoice call
```

Comment:
Why not do the `removeInvoice` in `SwapRecovery`?

Reply:
Hm we could, my thinking though is that if we haven't started sending payment yet and are just in the acceptance phase, there are no funds that can possibly be lost and no uncertainty as to whether it's safe to cancel invoices.